walkers/src/extras/images.rs

```diff
@@ -0,0 +1,236 @@
+        Self { images }
+    }
+}
+
+impl Plugin for Images {
+    fn draw(&self, painter: egui::Painter, projector: &crate::Projector) {
+        for image in &self.images {
+            let screen_position = projector.project(image.position);
+            let map_rect = painter.clip_rect();
+            let texture = &image.texture;
+
+            let [w, h] = texture.size();
+            let w = w as f32;
+            let h = h as f32;
+            let mut rect = map_rect.translate(screen_position);
+
+            rect.min.x -= w / 2.0;
+            rect.min.y -= h / 2.0;
+
+            rect.max.x = rect.min.x + w;
+            rect.max.y = rect.min.y + h;
+
+            let skip = (rect.max.x < map_rect.min.x)
+                | (rect.max.y < map_rect.min.y)
+                | (rect.min.x > map_rect.max.x)
+                | (rect.min.y > map_rect.max.y);
+
+            if skip {
+                continue;
+            }
+
+            painter.image(
+                texture.id(),
+                rect,
+                Rect::from_min_max(pos2(0.0, 0.0), pos2(1.0, 1.0)),
+                Color32::WHITE,
+            );
+        }
+    }
+}
+
+trait ByPixel {
+    fn get_pixel(&self, x: usize, y: usize) -> Option<Color32>;
+    fn set_pixel(&mut self, x: usize, y: usize, color: Color32);
+    fn index_to_xy(&self, i: usize) -> (usize, usize);
+}
+
+impl ByPixel for ColorImage {
+    fn get_pixel(&self, x: usize, y: usize) -> Option<Color32> {
+        let [w, h] = self.size;
+        if x >= w || y >= h {
+            return None;
+        }
+        Some(self.pixels[x + w * y])
+    }
+
+    fn set_pixel(&mut self, x: usize, y: usize, color: Color32) {
+        let w = self.size[0];
+        self.pixels[x + w * y] = color;
+    }
+
+    fn index_to_xy(&self, i: usize) -> (usize, usize) {
+        let w = self.size[0];
+        let y = i / w;
+        let x = i % w;
+        (x, y)
+    }
+}
+
+impl Texture {
+    /// Construct new texture
+    /// ⚠️ Make sure to only call this ONCE for each image, i.e. NOT in your main GUI code.
+    /// The call is NOT immediate safe.
+    pub fn new(ctx: Context, uri: &str, img: ColorImage) -> Self {
+        let texture = ctx.load_texture(uri, img.clone(), Default::default());
+
+        Self {
+            img: img.clone(),
+            scaled_img: img,
+            texture: texture,
+            x_scale: 1.0,
+            y_scale: 1.0,
+            angle: 0.0,
+        }
+    }
+
+    /// Same as [egui::TextureHandle::id]
+    /// (https://docs.rs/egui/latest/egui/struct.TextureHandle.html#method.id)
+    #[inline(always)]
+    pub fn id(&self) -> TextureId {
+        self.texture.id()
+    }
+
+    /// Same as [egui::TextureHandle::size] (https://docs.rs/egui/latest/egui/struct.TextureHandle.html#method.size)
+    #[inline(always)]
+    pub fn size(&self) -> [usize; 2] {
+        self.texture.size()
+    }
+
+    /// Scale texture.
+    pub fn scale(&mut self, x_val: f32, y_val: f32) {
```

Comment:
Scaling is implemented somewhere in egui as well.

https://docs.rs/egui/latest/egui/widgets/struct.Image.html

Reply:
Redone in new commit.